### PR TITLE
[GOVUKAPP-667] Use search api v2

### PR DESCRIPTION
**Pull request by 1pretz1**
Switches to search api v2. We now pull in the search url from the remote config: https://github.com/alphagov/govuk-mobile-backend-config/pull/18. We expect the v2 URL to change (host + path) and so being able to swap on the fly is needed. Also has the advantage of being able to switch to search v1 if v2 is down. Code has been updated to support both endpoints.

We now handle blank description for a search item too and only render a title and link for these items. Previously an error was thrown if a search result included a blank description.

[Jira ticket](https://govukverify.atlassian.net/jira/software/c/projects/GOVUKAPP/boards/559?selectedIssue=GOVUKAPP-667)

### CODE REVIEW

File: Production/govuk_ios/Services/AppConfigService.swift [hunk 1]
```diff
@@ -29,17 +31,18 @@ public final class AppConfigService: AppConfigServiceInterface {
...
-            }
-        )
-
-        appConfigServiceClient.fetchAppConfig(
             completion: { [weak self] result in
                 self?.handleResult(result)
                 completion()
             }
         )
+//
```
Comment (joshdubey) [inline Production/govuk_ios/Services/AppConfigService.swift:39]: please remove commented code

**Review by joshdubey — CHANGES_REQUESTED** [review]

**Review by batesyuk — APPROVED** [review]

File: Production/govuk_ios/Services/AppConfigService.swift (hunk 1 above)
Comment (1pretz1) [reply Production/govuk_ios/Services/AppConfigService.swift]: Comment no longer applicable

**Review by batesyuk — APPROVED** [review]

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/Extensions/Container/Container+APIClientsTests.swift [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import Testing
+
+import Factory
+
+@testable import govuk_ios
+
+@Suite
+struct Container_APIClientTests {
+    @Test
+    func searchAPIClient_returnsExpectedResult() async {
```
Comment (joshdubey) [inline Tests/govuk_ios/govuk_ios_unit_tests/Specs/Extensions/Container/Container+APIClientsTests.swift:11]: I'd be inclined to change the name of this test to searchAPIClient_createsExpectedRequest.  There is nothing about the result being tested.

**Review by joshdubey — APPROVED** [review]

**Review by raphaelolu — CHANGES_REQUESTED** [review]
Could we please add a description to the PR , this will make it easier in the cases where other devs need to look back at this

File: Tests/govuk_ios/govuk_ios_unit_tests/Specs/Extensions/Container/Container+APIClientsTests.swift (hunk 2 above)
Comment (1pretz1) [reply Tests/govuk_ios/govuk_ios_unit_tests/Specs/Extensions/Container/Container+APIClientsTests.swift]: Updated 👍 Thanks!

**Comment by 1pretz1** [comment]
The issues have been resolved @raphaelolu (false positives for our use case).

File: Production/govuk_ios/Services/AppConfigService.swift [hunk 3]
```diff
@@ -64,6 +66,16 @@ public final class AppConfigService: AppConfigServiceInterface {
                 new
             }
         )
+        updateSearch(urlString: config.searchApiUrl)
+    }
+
+    private func updateSearch(urlString: String?) {
+        guard let urlString = urlString,
+              let url = URL(string: urlString),
+              let components = URLComponents(url: url, resolvingAgainstBaseURL: true)
```
Comment (raphaelolu) [inline Production/govuk_ios/Services/AppConfigService.swift:75]: indenting

File: Tests/govuk_ios/govuk_ios_unit_tests/Mocks/Networking/MockURLProtocol.swift [hunk 4]
```diff
@@ -10,12 +10,17 @@ class MockURLProtocol: URLProtocol {
...
     }
 
-//    static var requestHandler: ((URLRequest) -> (HTTPURLResponse, Data?, Error?)?)?
     static var requestHandlers: [String: ((URLRequest) -> (HTTPURLResponse, Data?, Error?)?)] = [:]
 
     override func startLoading() {
-        guard let url = request.url?.absoluteString,
-              let handler = MockURLProtocol.requestHandlers[url] 
+        guard let requestUrl = request.url,
+              var components = URLComponents(url: requestUrl, resolvingAgainstBaseURL: true)
```
Comment (raphaelolu) [inline Tests/govuk_ios/govuk_ios_unit_tests/Mocks/Networking/MockURLProtocol.swift:17]: indenting

File: Production/govuk_ios/Services/AppConfigService.swift (hunk 3 above)
Comment (1pretz1) [reply Production/govuk_ios/Services/AppConfigService.swift:75]: Discussed offline, keeping original indenting

File: Tests/govuk_ios/govuk_ios_unit_tests/Mocks/Networking/MockURLProtocol.swift (hunk 4 above)
Comment (1pretz1) [reply Tests/govuk_ios/govuk_ios_unit_tests/Mocks/Networking/MockURLProtocol.swift:17]: Discussed offline, keeping original indenting

**Pull request merged**